ApiDoctor.Validation/Http/HttpParser.cs

```diff
@@ -23,115 +23,118 @@
+        ///     Converts a raw HTTP request into an HttpWebRequest instance.
         /// </summary>
         /// <param name="requestString"></param>
         /// <returns></returns>
         public HttpRequest ParseHttpRequest(string requestString)
         {
             if (string.IsNullOrWhiteSpace(requestString))
-            {
                 throw new ArgumentException("requestString was empty or whitespace only. Not a valid HTTP request.");
-            }
 
-            StringReader reader = new StringReader(requestString);
+            var firstNewLine = requestString.IndexOf(Environment.NewLine, StringComparison.InvariantCultureIgnoreCase);
+            if ($"{requestString.ElementAtOrDefault(firstNewLine)}{requestString.ElementAtOrDefault(firstNewLine + 1)}" == Environment.NewLine)
+            {
+                requestString = requestString.Remove(firstNewLine, 2);
+            }
+            var reader = new StringReader(requestString);
             string line;
-            ParserMode mode = ParserMode.FirstLine;
-
-            HttpRequest request = new HttpRequest();
+            var bodyBrace = string.Empty;
+            var mode = ParserMode.FirstLine;
 
-            while( (line = reader.ReadLine()) != null)
+            var request = new HttpRequest();
+            while ((line = reader.ReadLine()) != null)
             {
+                line = line.Trim();
+
                 switch (mode)
                 {
                     case ParserMode.FirstLine:
+
                         var components = line.Split(' ');
-                        if (components.Length < 2) 
-                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line.");
-                        if (components.Length > 3)
-                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line (more than the expected 3 components)");
+                        var componentsLength = components.Length;
 
+                        if (componentsLength < 2)
+                            throw new HttpParserRequestException(
+                                "RequestString does not contain a proper HTTP request first line.");
                         if (components[0].StartsWith("HTTP/"))
                             throw new HttpParserRequestException("RequestString contains an HTTP response.");
 
                         request.Method = components[0];
-                        request.Url = components[1];
-
-                        if (components.Length >= 3)
-                        {
-                            request.HttpVersion = components[2];
-                        }
-                        else
-                        {
-                            request.HttpVersion = "HTTP/1.1";
-                        }
+                        request.Url = componentsLength > 3 ? line.Replace(request.Method, string.Empty) : components[1];
```

Comment:
what's this doing?

Reply:
I had tried to handle edge cases arising from doc write mistakes. I decided to abandon this and force them to fix the docs.